oci/table_oci_core_nat_gateway.go

```diff
@@ -19,8 +19,9 @@ func tableCoreNatGateway(_ context.Context) *plugin.Table {
 		Name:        "oci_core_nat_gateway",
 		Description: "OCI Core Nat Gateway",
 		Get: &plugin.GetConfig{
-			KeyColumns: plugin.SingleColumn("id"),
-			Hydrate:    getCoreNatGateway,
+			KeyColumns:        plugin.SingleColumn("id"),
+			ShouldIgnoreError: isNotFoundError([]string{"400", "404"}),
```

Comment:
Please move the changes made in the `.go` files into a separate PR since they refer to a separate bug altogether